edgexfoundry/app-functions-sdk-go
# refactor!: Update config for removal of SecretStore from services' configuration file

BREAKING CHANGE: SecretStore config no longer in service configuration file. Changes must be done via use of environment variable overrides of default values.

This PR is dependent on https://github.com/edgexfoundry/go-mod-bootstrap/pull/414

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  TBD

## Testing Instructions
Run secure EdgeX stack with `app-new-service` added to appropriate env vars
Build template app  ( `app-new-service`)  from this branch
Run service local with following command:
```
sudo EDGEX_SECURITY_SECRET_STORE=true WRITABLE_LOGLEVEL=DEBUG ./ `app-new-service` -cp -r
```
Verify service bootstraps without errors (i.e. connects to Consul and MessageBus securely) 
Verify config is loaded from file and pushed to Consul w/o errors
Verify `SecretStore` config no longer in service's config in Consul
Restart service
Verify config is loaded from consul only, i.e. not first loaded from file.

## New Dependency Instructions (If applicable)
